pkg/forklift-api/services/tls-certificate.go

```diff
@@ -17,6 +18,11 @@ func serveTlsCertificate(resp http.ResponseWriter, req *http.Request, client cli
 			Data: map[string][]byte{"insecureSkipVerify": []byte("true")},
 		}
 		if cacert, err := util.GetTlsCertificate(url, secret); err == nil {
+			cacert, err := x509.ParseCertificate(cacert.Raw)
+			if err != nil {
+				msg := fmt.Sprintf("failed to parse certificate: %v", err)
+				http.Error(resp, msg, http.StatusInternalServerError)
```

Comment:
and we should break here without proceeding to `resp.Write` below
but as we anyway need to change this PR, let me move it to the utl.GetTlsCertificate function, it will simplify the code a bit